packages/labs/react/src/create-component.ts

```diff
@@ -106,6 +106,24 @@ type EventProps<R extends Events> = {
     : (e: Event) => void;
 };
 
+type ElementChildren<C> = Record<keyof C, React.ReactNode>;
+
+const createSlottedChild = (
+  React: typeof ReactModule,
+  slot: string,
+  children: React.ReactNode
+) => {
+  return React.createElement(
+    'div',
+    {
+      style: {display: 'contents'},
+      key: slot,
+      slot,
+    },
+    children
```

Comment:
This is arguable and we should sync up on this before changing the code, but I think it's preferable to set slot attributes where possible rather than to create a wrapper DOM node.

However, if a wrapper node is needed, probably we should just make one and not set slots. So, this would involve iterating the children until you find a non-element node (text nodes or component references): if so making the wrapper; if not, cloneElement and set slots.

Reply:
I believe React.ReactNode can be any value that react will render: string array ref node element. Children are passed through a single property name in react / jsx (or as "children"). So when a `child` is assed to `createSlottedChild` it wraps the entire prop to a slot via one wrap.

I figured this was less operations for the same behavior, choosing to wrap the entire children prop in a single function than search and create multiple. If the CSS `display contents` let's us avoid styling I figured this would be the most direct way to slot quickly and cheaply

A clarifying question: Are we expect a user to pass in some value than a React.ReactNode?

Reply:
Creating the wrapper DOM node itself is likely the most expensive part. If we can avoid it and just set slot attributes, we probably should. I wanted to explore what was possible via their API and so made [this experiment](https://stackblitz.com/edit/react-ts-pfwb7c?file=slots.ts,App.tsx) which might be useful.